Mathlib/Data/Set/Image.lean

```diff
@@ -518,6 +518,20 @@ theorem image_preimage_eq {f : α → β} (s : Set β) (h : Surjective f) : f ''
     ⟨y, (e.symm ▸ hx : f y ∈ s), e⟩
 #align set.image_preimage_eq Set.image_preimage_eq
 
+/-- Without this theorem, `image_subset_iff` and `image_preimage_eq` create a simp confluence issue.
+ -/
```

Comment:
I don't think those three docstrings are good docstrings. I think they should be regular comments. This information is all about the `simp` attribute and this is not what users want to see as documentation.

Reply:
Rather than saying there's a confluence issue, could you mention concretely what the confluence issue is? In particular, it would be helpful to see a term that simplifies in two ways depending on the order the theorems are applied, and also what these terms both get simplified into given this new simp theorem.

Reply:
I added machine-checked examples of the confluence issue and moved the docstrings to document the examples instead of the theorems, which seems more appropriate. I don't think there's precedent for this kind of confluence regression test in the middle of a file, but this seemed like a better option than adding just a more explicit comment since now if these lemmas become unnecessary we'll find out.

Reply:
You shouldn't put tests/examples in mathlib files, put them in `tests/` if needed.

Reply:
Okay. I moved them to `tests/simp_confluence.lean`. Now there is no comment on these lemmas, which might be fine. I figure if someone is wondering about what their purpose is, one of the first things they will do is "Go to references" which will turn up these tests.